src/main/java/commands/EventCommand.java

```diff
@@ -0,0 +1,141 @@
+        int endDateTimeWords = 0;
+        for (String word : commandWords) {
+            if (word.equalsIgnoreCase("/from")) {
+                isSeeingStartDateTime = true;
+                continue;
+            }
+            if (word.equalsIgnoreCase("/to")) {
+                isSeeingStartDateTime = false;
+                isSeeingEndDateTime = true;
+                continue;
+            }
+            if (isSeeingStartDateTime) {
+                startDateTimeWords += 1;
+                startDateTime.append(word)
+                        .append(" ");
+            }
+            if (isSeeingEndDateTime) {
+                endDateTimeWords += 1;
+                endDateTime.append(word)
+                        .append(" ");
+            }
+        }
+
+        if (description.isEmpty()) {
+            throw new BrockException("Description is missing!");
+        }
+        if (startDateTime.isEmpty()) {
+            throw new BrockException("Missing start date! Remember it is specified after /from!");
+        }
+        if (endDateTime.isEmpty()) {
+            throw new BrockException("Missing end date! Remember it is specified after /to!");
+        }
+        if (startDateTimeWords != endDateTimeWords) {
+            throw new BrockException("Both start and end dates must either include or exclude a time!");
+        }
+
+        String[] startDateTimeValues = Utility.validateDateTime(startDateTime.toString(),
+                startDateTimeWords, Utility.Context.START);
+        String[] endDateTimeValues = Utility.validateDateTime(endDateTime.toString(),
+                endDateTimeWords, Utility.Context.END);
+        if (startDateTimeWords == 1) {
+            return new Event(description.toString(),
+                    startDateTimeValues[0],
+                    endDateTimeValues[0]);
+        } else {
+            return new Event(description.toString(),
+                    startDateTimeValues[0],
+                    startDateTimeValues[1],
+                    endDateTimeValues[0],
+                    endDateTimeValues[1]);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>
+     * Chatbot checks if event command is valid.
+     * If so, it creates a {@code Event} object.
+     * Adds it to {@code tasks}, writes it to save file.
+     * Displays a response indicating it has added the event task.
+     * </p>
+     *
+     * @throws BrockException If event command is invalid
+     */
+    @Override
+    public void execute(Ui ui, Storage storage, TaskList tasks) throws BrockException {
+        Task eventTask = createEvent();
+        tasks.addToList(eventTask);
+        ui.displayResponse("Got it. I've added this task:\n"
+                + "  "
+                + tasks.getTaskDetails(eventTask)
+                + '\n'
+                + tasks.getTasksSummary());
+
+        // Update the save file
+        storage.writeToFile(tasks.numTasks()
+                        + ". "
```

Comment:
I think the auto-indentations on these lines may not be correct, would be good to fix! 😄 
I noticed a similar problem in your TodoCommand.java around lines 71-74, so be sure to check for those.

Reply:
Based on the style guide it says indentation of wrapped lines is 8 spaces (ie: 2x the usual 4 spaces), so i think this is correct? 😅

Reply:
Oh, it shows as 16 spaces on my end...but as long as it looks right on your machine, all good!
<img width="490" alt="image" src="https://github.com/user-attachments/assets/05d98948-4642-4ee0-b64b-bb0e06584a4d">

Reply:
EH my bad i was looking at the wrong part, yea i fixed it now, thankss 👍